internal/txlib/add_remote.go

```diff
@@ -0,0 +1,129 @@
+package txlib
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/transifex/cli/internal/txlib/config"
+	"github.com/transifex/cli/pkg/jsonapi"
+	"github.com/transifex/cli/pkg/txapi"
+)
+
+func AddRemoteCommand(
+	cfg *config.Config,
+	api *jsonapi.Connection,
+	projectUrl,
+	fileFilter string,
+	minimumPerc int,
+) error {
+	parts := strings.Split(projectUrl, "/")
+	for parts[len(parts)-1] == "dashboard" || parts[len(parts)-1] == "" {
+		parts = parts[:len(parts)-1]
+	}
+	if len(parts) < 2 {
+		return fmt.Errorf("invalid project URL '%s'", projectUrl)
+	}
+	organizationSlug := parts[len(parts)-2]
+	projectSlug := parts[len(parts)-1]
+
+	project, err := txapi.GetProjectById(
+		api,
+		fmt.Sprintf("o:%s:p:%s", organizationSlug, projectSlug),
+	)
+	if err != nil {
+		return fmt.Errorf("unable to fetch project: %s", err)
+	}
+	resources, err := txapi.GetResources(api, project)
+	if err != nil {
+		return fmt.Errorf("unable to fetch resources: %s", err)
+	}
+	organization := &jsonapi.Resource{
+		API:  api,
+		Type: "organizations",
+		Id:   fmt.Sprintf("o:%s", organizationSlug),
+	}
+	i18nFormats, err := txapi.GetI18nFormats(api, organization)
+	if err != nil {
+		return fmt.Errorf("unable to fetch i18n formats: %s", err)
+	}
```

Comment:
Apart from error, we also need to check if we have any content in these requests.